muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspacenotes/WorkspaceNoteController.java

```diff
@@ -0,0 +1,105 @@
+package fi.otavanopisto.muikku.plugins.workspacenotes;
+
+import java.util.Comparator;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceNote;
+import fi.otavanopisto.muikku.plugins.workspacenotes.dao.WorkspaceNoteDAO;
+
+public class WorkspaceNoteController {
+  
+  @Inject
+  private WorkspaceNoteDAO workspaceNoteDAO;
+  
+  public List<WorkspaceNote> listByOwnerAndArchived(Long owner, Boolean archived) {
+    List<WorkspaceNote> workspaceNotes = workspaceNoteDAO.listByOwnerAndArchived(owner, archived);
+    workspaceNotes.sort(Comparator.comparing(WorkspaceNote::getOrderNumber));
+    return workspaceNotes; 
+  }
+  
+  public List<WorkspaceNote> listByWorkspaceAndOwnerAndArchived(Long workspaceEntityId, Long owner, Boolean archived){
+    List<WorkspaceNote> workspaceNotes = workspaceNoteDAO.listByOwnerAndWorkspaceAndArchived(owner, workspaceEntityId, archived);
+    workspaceNotes.sort(Comparator.comparing(WorkspaceNote::getOrderNumber));
+    return workspaceNotes;
+  }
+  
+  public WorkspaceNote createWorkspaceNote(Long owner, String title, String note, Long workspaceId) {
+    
+    Integer maximumOrderNumber = workspaceNoteDAO.getMaximumOrderNumberByOwnerAndWorkspace(workspaceId, owner);
+    
+    if (maximumOrderNumber == null) {
+      maximumOrderNumber = 0;
+    } else {
+      ++maximumOrderNumber;
+    }
+    return workspaceNoteDAO.create(
+        owner, 
+        title,
+        note, 
+        workspaceId, 
+        maximumOrderNumber);
+  }
+  
+  public WorkspaceNote updateWorkspaceNote(WorkspaceNote workspaceNote, String title, String note) {
+    return workspaceNoteDAO.update(workspaceNote, title, note);
+  }
+  
+  public WorkspaceNote archive(WorkspaceNote workspaceNote) {
+    return workspaceNoteDAO.setArchived(workspaceNote, Boolean.TRUE);
+  }
+  
+  public WorkspaceNote findWorkspaceNoteById(Long id) {
+    return workspaceNoteDAO.findById(id);
+  }
+  
+  public WorkspaceNote updateOrderNumber(WorkspaceNote workspaceNote, Integer newOrderNumber) {
+
+    return workspaceNoteDAO.updateOrderNumber(workspaceNote, newOrderNumber);
+    
+  }
+  
+  /**
+   * Updates the order numbers of workspace notes so that
+   * <code>workspaceNote</code> appears above <code>referenceNote</code>.
+   * 
+   * @param workspaceNote
+   *          The workspace note to be moved
+   * @param referenceNote
+   *          The workspace note above which <code>workspaceNote</code> is moved
+   * 
+   * @return The updated workspace note
+   */
+  public WorkspaceNote moveAbove(WorkspaceNote workspaceNote, WorkspaceNote referenceNode) {
```

Comment:
There probably would be a way to do this within DAO with CriteriaUpdate instead of looping, but ok like this.